src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/PreferDictionaryTryMethodsOverContainsKeyGuardAnalyzer.cs

```diff
@@ -556,9 +556,9 @@ private static bool IsSameReferenceOperation(IOperation? sourceReference, IOpera
                 case IParameterReferenceOperation source when targetReference is IParameterReferenceOperation target:
                     return target.Parameter.Equals(source.Parameter, SymbolEqualityComparer.Default);
                 case IFieldReferenceOperation source when targetReference is IFieldReferenceOperation target:
-                    return target.Field.Equals(source.Field, SymbolEqualityComparer.Default);
+                    return target.Field.Equals(source.Field, SymbolEqualityComparer.Default) && AreInstancesEqual(source, target);
                 case IPropertyReferenceOperation source when targetReference is IPropertyReferenceOperation target:
-                    return target.Property.Equals(source.Property, SymbolEqualityComparer.Default);
+                    return target.Property.Equals(source.Property, SymbolEqualityComparer.Default) && AreInstancesEqual(source, target);
```

Comment:
I would expect to add something like `&& source.Instance == target.Instance`, but it doesn't work. 

Comparing string representation looks bit weird, though don't see any other way